### PR TITLE
Removed tank dispenser UI

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

Removed tank dispenser UI, making the tanks dispense on left and right clicks instead.

image

## Why It's Good For The Game

Less clicks. 

## Changelog

:cl:
del: Removed tank dispenser UI to use left/right clicks instead
/:cl:

### CODE REVIEW

File: code/game/objects/structures/tank_dispenser.dm [hunk 1]
```diff
@@ -34,6 +35,22 @@
...
+	. = ..()
+	if (plasmatanks)
+		dispense(/obj/item/tank/internals/plasma, user)
+		plasmatanks--
+		update_appearance()
+		return
+
+/obj/structure/tank_dispenser/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if (oxygentanks)
```
Comment (LemonInTheDark) [inline code/game/objects/structures/tank_dispenser.dm:48]: early return

**Comment by alien3301** [comment]
Make it tell you how many tanks it has on examine if it doesn't do that already

File: code/game/objects/structures/tank_dispenser.dm [hunk 2]
```diff
@@ -34,6 +35,22 @@
 		if(5 to TANK_DISPENSER_CAPACITY)
 			. += "plasma-5"
 
+/obj/structure/tank_dispenser/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if (plasmatanks)
+		dispense(/obj/item/tank/internals/plasma, user)
+		plasmatanks--
+		update_appearance()
+		return
```
Comment (LemonInTheDark) [inline code/game/objects/structures/tank_dispenser.dm:44]: also this return is redundant, did you mean to put something here?

Comment (MTandi) [reply code/game/objects/structures/tank_dispenser.dm:44]: It will continue the attack chain if I won't return.

**Comment by MrMelbert** [comment]
I was looking at this UI recently and thought "man I'd be sad if this UI was removed it's so soulful"

**Comment by MTandi** [comment]
> I was looking at this UI recently and thought "man I'd be sad if this UI was removed it's so soulful"

I'm sorry

File: code/game/objects/structures/tank_dispenser.dm (hunk 1 above)
Comment (MTandi) [reply code/game/objects/structures/tank_dispenser.dm:48]: `attack_hand` will be called if I won't return in `attack_hand_secondary`

**Comment by YuliceTheDruid** [comment]
Can you make Oxy tanks left click :). I use those WAY more often than a plasma tank.

**Comment by MTandi** [comment]
> Can you make Oxy tanks left click :). I use those WAY more often than a plasma tank.

I made them correspond the icon. Oxy tanks are on the right on the icon.

File: code/game/objects/structures/tank_dispenser.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/game/objects/structures/tank_dispenser.dm:44]: so this is `return NONE`? can you do that explicitly please?

File: code/game/objects/structures/tank_dispenser.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/game/objects/structures/tank_dispenser.dm:48]: ok so please early return

File: code/game/objects/structures/tank_dispenser.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/game/objects/structures/tank_dispenser.dm:44]: This doesn't make any sense, `return` hands back . by default, just the same way putting nothing at the end of a proc does
image

File: code/game/objects/structures/tank_dispenser.dm [hunk 3]
```diff
@@ -34,6 +35,25 @@
...
+	dispense(/obj/item/tank/internals/plasma, user)
+	plasmatanks--
+	update_appearance()
+
+/obj/structure/tank_dispenser/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if (!oxygentanks)
+		balloon_alert(user, "no oxygen tanks!")
+		return
```
Comment (LemonInTheDark) [inline code/game/objects/structures/tank_dispenser.dm:51]: should this also SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN?

File: code/game/objects/structures/tank_dispenser.dm [hunk 4]
```diff
@@ -52,67 +72,34 @@
...
-
 /obj/structure/tank_dispenser/atom_deconstruct(disassembled = TRUE)
 	for(var/X in src)
 		var/obj/item/I = X
 		I.forceMove(loc)
 	new /obj/item/stack/sheet/iron (loc, 2)
 
+/obj/structure/tank_dispenser/examine(mob/user)
+	. = ..()
+	. += span_notice("It has <b>[plasmatanks]</b> plasma tank\s and <b>[oxygentanks]</b> oxygen tank\s left.")
```
Comment (ShizCalev) [inline code/game/objects/structures/tank_dispenser.dm:96]:
```suggestion
	. = ..()
	if(plasmatanks && oxygentanks)
		. += span_notice("It has <b>[plasmatanks]</b> plasma tank\s and <b>[oxygentanks]</b> oxygen tank\s left.")
	else if(plasmatanks || oxygentanks))
		. += span_notice("It has <b>[plasmatanks ? "[plasmatanks]</b> plasma" : [oxygentanks]</b> oxygen" tank\s left.)
```

File: code/game/objects/structures/tank_dispenser.dm [hunk 5]
```diff
@@ -52,67 +72,37 @@
...
 		var/obj/item/I = X
 		I.forceMove(loc)
 	new /obj/item/stack/sheet/iron (loc, 2)
 
+/obj/structure/tank_dispenser/examine(mob/user)
+	. = ..()
+	if(plasmatanks && oxygentanks)
+		. += span_notice("It has <b>[plasmatanks]</b> plasma tank\s and <b>[oxygentanks]</b> oxygen tank\s left.")
+	else if(plasmatanks || oxygentanks))
+		. += span_notice("It has <b>[plasmatanks ? "[plasmatanks]</b> plasma" : [oxygentanks]</b> oxygen" tank\s left.)
```
Comment (ShizCalev) [inline code/game/objects/structures/tank_dispenser.dm:99]:
```suggestion
		. += span_notice("It has <b>[plasmatanks ? "[plasmatanks]</b> plasma" : "[oxygentanks]</b> oxygen" tank\s left.)
```

File: code/game/objects/structures/tank_dispenser.dm [hunk 6]
```diff
@@ -52,67 +72,37 @@
...
 	for(var/X in src)
 		var/obj/item/I = X
 		I.forceMove(loc)
 	new /obj/item/stack/sheet/iron (loc, 2)
 
+/obj/structure/tank_dispenser/examine(mob/user)
+	. = ..()
+	if(plasmatanks && oxygentanks)
+		. += span_notice("It has <b>[plasmatanks]</b> plasma tank\s and <b>[oxygentanks]</b> oxygen tank\s left.")
+	else if(plasmatanks || oxygentanks))
```
Comment (ShizCalev) [inline code/game/objects/structures/tank_dispenser.dm:98]:
```suggestion
	else if(plasmatanks || oxygentanks)
```

**Comment by MTandi** [comment]
@ShizCalev if we do this, then it won't say that it has 0 tanks, which is not what I want. You should be able to tell by examine that there are no plasma/oxygen tanks.

**Comment by LemonInTheDark** [comment]
) moment

**Pull request merged**